src/routes/auth/login.test.ts

```diff
@@ -0,0 +1,161 @@
+import { login } from './login'
+import jws from 'jws'
+import { createSpyObj } from '../../test/helper/jest'
+import { Environment } from '../../utils/Environment'
+import * as val from 'express-validator'
+jest.mock('express-validator')
+
+describe('Check login', () => {
+  let resSpy
+  let req
+
+  beforeEach(() => {
+    resSpy = createSpyObj('Response', ['status', 'json', 'end', 'send'])
+    req = {
+      body: {
+        username: 'admin',
+        password: 'Passw0rd'
+      }
+    }
+    resSpy.status.mockReturnThis()
+    resSpy.json.mockReturnThis()
+    resSpy.send.mockReturnThis()
+    jest.spyOn(val, 'validationResult').mockImplementation(() => {
+      return {
+        isEmpty: jest.fn().mockReturnValue(true),
+        array: jest.fn().mockReturnValue([{ test: 'error' }])
+      } as any
+    })
+  })
+  it('should pass with correct user and password', async () => {
+    Environment.Config = {
+      web_admin_user: 'admin',
+      web_admin_password: 'Passw0rd',
+      jwt_expiration: 1440,
+      jwt_issuer: 'fake',
+      jwt_secret: 'my_secret'
+    } as any
+    await login(req, resSpy)
+    expect(resSpy.status).toHaveBeenCalledWith(200)
+  })
+  it('should fail with incorrect user', async () => {
+    Environment.Config = {
+      web_admin_user: 'fake',
+      web_admin_password: 'Passw0rd',
+      jwt_expiration: 1440,
+      jwt_issuer: 'fake',
+      jwt_secret: 'my_secret'
+    } as any
+    await login(req, resSpy)
+    expect(resSpy.status).toHaveBeenCalledWith(401)
+  })
+  it('should pass with lowecase user', async () => {
+    Environment.Config = {
+      web_admin_user: 'ADMIN',
+      web_admin_password: 'Passw0rd',
+      jwt_expiration: 1440,
+      jwt_issuer: 'fake',
+      jwt_secret: 'my_secret'
+    } as any
+    await login(req, resSpy)
+    expect(resSpy.status).toHaveBeenCalledWith(200)
+  })
+  it('should pass with uppercase user', async () => {
+    req = {
+      body: {
+        username: 'ADMIN',
+        password: 'Passw0rd'
+      }
+    }
+    Environment.Config = {
+      web_admin_user: 'admin',
+      web_admin_password: 'Passw0rd',
+      jwt_expiration: 1440,
+      jwt_issuer: 'fake',
+      jwt_secret: 'my_secret'
+    } as any
+    await login(req, resSpy)
+    expect(resSpy.status).toHaveBeenCalledWith(200)
+  })
+  it('should pass with expected expiration', async () => {
+    jest.spyOn(jws, 'sign').mockImplementation(() => {
```

Comment:
while this works -- I was kind of thinking that we decrypt the returned token to ensure that it is correct as opposed to mocking the implementation.

Reply:
Done. Please let me know if that is the expected result